core/store/src/flat/store_helper.rs

```diff
@@ -82,6 +82,14 @@ pub fn remove_delta(store_update: &mut StoreUpdate, shard_uid: ShardUId, block_h
     store_update.delete(FlatStateColumn::DeltaMetadata.to_db_col(), &key);
 }
 
+pub fn remove_all_deltas(store_update: &mut StoreUpdate, shard_uid: ShardUId) {
+    let key_from = shard_uid.to_bytes();
+    let mut key_to = key_from;
+    key_to[7] += 1;
```

Comment:
what if `key_to[7]` is `u8::MAX_VALUE`? 
I suggest implementing `fn next_shard_prefix(cur: &[u8]) -> Vec<u8>` function which does `+1` with proper carry over handling. Then we can re-used it for flat state removal as well.

Reply:
I see. There would be no impact because we have < 256 shards, but I don't want it to suddenly fail at some point. Introduced `ShardUId::next_shard_prefix(bytes)` because it makes more sense as slice operation.

Side note: little-endian is quite annoying here, because I can't just increment shard id. Still we can rely on the fact that all shard uids are unique.